### PR TITLE
[Merged by Bors] - feat(Topology): clopen subsets of products of compact spaces are unions of clopen boxes

### PR DESCRIPTION
**Pull request by dagurtomas**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/ClopenBox.lean
Comment (jcommelin) [inline Mathlib/Topology/ClopenBox.lean:15]: Minor nitpick:
```suggestion
However, when one of the factors is compact, a clopen subset can be written as such a union.
```

File: Mathlib/Topology/ClopenBox.lean
Comment (jcommelin) [inline Mathlib/Topology/ClopenBox.lean:20]:
```suggestion
cartesian products of compact spaces (this is relevant to the theory of light profinite sets).
```

File: Mathlib/Topology/ClopenBox.lean
Comment (jcommelin) [inline Mathlib/Topology/ClopenBox.lean:167]: Can you golf this by merging the `rw`s into a `simp only`?

File: Mathlib/Topology/ClopenBox.lean
Comment (jcommelin) [inline Mathlib/Topology/ClopenBox.lean:19]: Do you mean "finitely many" or "countably many" (which is what you show at the end of the file, and which seems more related to the "light" story)?

File: Mathlib/Topology/ClopenBox.lean
Comment (jcommelin) [inline Mathlib/Topology/ClopenBox.lean:85]: This looks like another golf opportunity with `simp`?

File: Mathlib/Topology/ClopenBox.lean
Comment (jcommelin) [inline Mathlib/Topology/ClopenBox.lean:92]: And here?

File: Mathlib/Topology/ClopenBox.lean
Comment (jcommelin) [inline Mathlib/Topology/ClopenBox.lean:99]: And here?

**Review by jcommelin — COMMENTED** [review]
The big proof is pretty big... are there natural sublemmas that can be extracted?

**Comment by dagurtomas** [comment]
I extracted the fact that `U` is open as a sublemma

File: Mathlib/Topology/ClopenBox.lean
Comment (urkud) [inline Mathlib/Topology/ClopenBox.lean:31]:
```suggestion
theorem isOpen_setOf_singleton_prod_contained {V : Set Y} (hV : IsCompact V) {W : Set (X × Y)}
```

File: Mathlib/Topology/ClopenBox.lean
Comment (urkud) [inline Mathlib/Topology/ClopenBox.lean:42]: This lemma should go to `Topology.Constructions`.

File: Mathlib/Topology/ClopenBox.lean
Comment (urkud) [inline Mathlib/Topology/ClopenBox.lean:38]: This lemma should right after `ContinuousMap.curry`.

**Comment by urkud** [comment]
I golfed the proofs using some lemmas that were already in the library. Now we need a review from another maintainer (e.g., I'm not sure if the names I use are too verbose).

**Comment by dagurtomas** [comment]
These changes look great. Thanks @urkud!

**Comment by urkud** [comment]
LGTM but I contributed too much to merge it.
maintainer merge

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**
